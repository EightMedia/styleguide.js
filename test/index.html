<!DOCTYPE html><html><head><title>Styleguide</title><script type="text/javascript">
</script><style type="text/css">article,
aside,
details,
figcaption,
figure,
footer,
header,
hgroup,
menu,
nav,
section {
  display: block;
}
body {
  background: #f5f5f5;
  font-size: 15px;
  line-height: 22px;
  font-family: Helvetica, Arial, sans-serif;
}
.container {
  max-width: 1200px;
  margin: 0 auto;
  padding: 20px;
}
.nav {
  position: fixed;
  left: 0;
  top: 0;
  height: 100%;
  width: 50px;
  overflow-y: auto;
  overflow-x: hidden;
  background: #333;
  color: #fff;
  z-index: 100;
  -webkit-transition: width .3s;
  -moz-transition: width .3s;
  transition: width .3s;
}
.nav:hover {
  width: 350px;
}
.nav ul {
  padding: 0;
  list-style: none;
  margin: 15px 0;
  opacity: .3;
}
.nav:hover ul {
  opacity: 1;
}
.nav a {
  padding: 5px 20px;
  color: #999;
  text-decoration: none;
  display: block;
  white-space: nowrap;
}
.nav .active a {
  background: #444;
  color: #fff;
}
iframe {
  border: none;
  margin: 0;
  background: #fff;
  border-radius: 5px;
  border: solid 1px rgba(0, 0, 0, 0.1);
}
.section {
  margin: 30px 0;
}
.section:after {
  content: "";
  display: block;
  width: 50px;
  height: 0;
  border-top: solid 3px #ddd;
}
.guide {
  margin: 30px 0;
}
.guide h3 {
  margin: 0 0 5px 0;
}
.guide .description p {
  margin: 0 0 20px 0;
}
.guide .preview iframe {
  width: 100%;
}
.guide .source {
  display: none;
}
.source_css {
  display: none;
}
</style></head><body><div class="container"><h1>Styleguide Demo</h1><nav class="nav"><ul><li><a href="#Forms">Forms</a></li><li><a href="#Forms-Common">Forms - Common</a></li></ul></nav><div id="Forms" class="section"><h2>Forms</h2><div class="guide"><h3>Test</h3><div class="preview"><iframe></iframe></div><div class="source"><textarea>&lt;div class=&quot;test&quot;&gt;This is a test&lt;/div&gt; &lt;div class=&quot;test&quot;&gt;This is another test&lt;/div&gt;</textarea></div></div></div><div id="Forms-Common" class="section"><h2>Forms - Common</h2><div class="guide"><h3>Another test</h3><div class="preview"><iframe></iframe></div><div class="source"><textarea>&lt;input type=&quot;text&quot; class=&quot;test&quot;&gt;</textarea></div></div><div class="guide"><h3>Labels</h3><div class="description"><p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer ac diam interdum augue blandit eleifend. Donec egestas, augue nec malesuada ullamcorper, sapien urna sollicitudin tellus, vel mattis ligula leo id mi. In sodales ante sollicitudin, varius metus id, varius nunc. Aenean rhoncus nisl a quam eleifend, vitae volutpat erat vulputate. Nullam scelerisque elit sit amet massa egestas, eu adipiscing lorem porttitor.</p>
</div><div class="preview"><iframe></iframe></div><div class="source"><textarea>&lt;label class=&quot;test&quot;&gt;This is a test&lt;/label&gt;</textarea></div></div></div><textarea class="source_css">body {
  font: 18px Verdana;
}

/***
  title: Test
  section: Forms
  example:
    &lt;div class=&quot;test&quot;&gt;This is a test&lt;/div&gt;
    &lt;div class=&quot;test&quot;&gt;This is another test&lt;/div&gt;
***/
.test,
.test2,
.test2[fancy=true] {
  background: blue;
  color: #fff;
}


/***
  title: Another test
  section: Forms - Common
  example: &lt;input type=&quot;text&quot; class=&quot;test&quot;&gt;
***/
input.test {
  background: green;
}


/***
  title: Labels
  section: Forms - Common
  description:
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer ac diam
    interdum augue blandit eleifend. Donec egestas, augue nec malesuada ullamcorper,
     sapien urna sollicitudin tellus, vel mattis ligula leo id mi. In sodales ante
     sollicitudin, varius metus id, varius nunc. Aenean rhoncus nisl a quam eleifend,
     vitae volutpat erat vulputate. Nullam scelerisque elit sit amet massa egestas,
      eu adipiscing lorem porttitor.
  example:
    &lt;label class=&quot;test&quot;&gt;This is a test&lt;/label&gt;
***/
label.test {
  background: yellow;
}</textarea></div><script type="text/javascript">
(function() {
  var d, guide, guide_source, iframe, scaleIframes, source_css, stylesheet, _i, _len, _ref;

  source_css = document.querySelector('textarea.source_css').textContent;

  _ref = document.querySelectorAll('.guide');
  for (_i = 0, _len = _ref.length; _i < _len; _i++) {
    guide = _ref[_i];
    iframe = guide.querySelector('.preview iframe');
    guide_source = guide.querySelector('.source textarea').textContent;
    d = iframe.contentWindow.document;
    d.open();
    d.write("<!DOCTYPE html><html><head></head><body>" + guide_source + "</body></html>");
    stylesheet = d.createElement('style');
    stylesheet.innerHTML = source_css;
    d.querySelector('head').appendChild(stylesheet);
    d.close();
  }

  scaleIframes = function() {
    var frame, _j, _len1, _ref1, _results;
    _ref1 = document.querySelectorAll('.preview iframe');
    _results = [];
    for (_j = 0, _len1 = _ref1.length; _j < _len1; _j++) {
      frame = _ref1[_j];
      _results.push(frame.height = frame.contentWindow.document.querySelector('html').offsetHeight);
    }
    return _results;
  };

  scaleIframes();

  window.addEventListener('resize', scaleIframes);

}).call(this);
</script></body></html>