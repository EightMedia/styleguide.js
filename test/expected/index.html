<!DOCTYPE html><html><head><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><title>Styleguide</title><style>@import url(http://fonts.googleapis.com/css?family=Open+Sans:400,600,700);


body {
  background: #fbfbfb;
  font-size: 15px;
  line-height: 22px;
  font-family: 'Open Sans', sans-serif;
  color: #222;
}

code {
  display: inline-block;
  padding: 2px 4px;
  color: green;
  background-color: #f7f7f9;
  border: 1px solid #e1e1e8;
  font-size: 12px;
  border-radius: 2px;
  font-family: Monaco, Menlo, Consolas, "Courier New", monospace;
}

.generated {
  font-size: 11px;
  color: #999;
  margin-bottom: 2em;
}

.container {
  max-width: 1200px;
  margin: 0 auto;
  padding: 20px 20px 20px 70px;
}


/* headings */
h1, h2, h3, h4, h5{
    font-size: 1em;
    line-height: 1;
    font-weight: normal;
    margin: 0 0 .5em 0;
}
.title-large{
  font-size: 2.5em;
  font-weight: 700;
  margin: 0 0 .2em 0;
}
.title-medium{
  font-size: 1.4em;
  font-weight: 600;
  text-transform: uppercase;
}
.title-small{
  font-size: 1.1em;
  font-weight: 600;
}


/* divider */
hr{
  display: block;
  height: 1px;
  border: 0;
  border-top: 1px solid #c9c9c9;
  margin: 0;
  padding: 0;
  clear: both;
}
hr.small{
  margin: 2em 0;
  width: 50px;
}


/* index */
.index{}

  .index ul{
    margin: 0;
    padding: 0;
    list-style: none;
  }
    .index a{
      color: darkslateblue;
      text-decoration: none;
    }
    .index a:hover{
      text-decoration: underline;
    }

    .index ul ul{
      margin-left: 1em;
    }


/* navigation */
.nav {
  position: fixed;
  left: 0;
  top: 0;
  height: 100%;
  width: 50px;
  overflow-y: auto;
  overflow-x: hidden;
  background: #222;
  border-right: 1px solid #c9c9c9;
  z-index: 100;
  -webkit-transition: width .3s;
  -moz-transition: width .3s;
  transition: width .3s;
}

.nav:hover {
  width: 350px;
}

  .nav ul {
    padding: 0;
    list-style: none;
    margin: 15px 0;
    opacity: .3;
  }

  .nav:hover ul {
    opacity: 1;
  }

  .nav a {
    padding: 5px 20px;
    color: #fefefe;
    text-decoration: none;
    display: block;
    white-space: nowrap;
  }

  .nav a:hover {
    text-decoration: underline;
  }



/* section */
.section {
  margin-top: 75px;
}
  .section > hr{
    margin-top: 75px;
  }


  .section:last-of-type{
    border: none;
    margin-bottom: 0;
    padding-bottom: 0;
  }



/* guide */
.guide {
  margin: 30px 0;
}

  .guide h3:before{
    margin-right: .3em;
    display: inline-block;
    -webkit-border-radius: 50%;
    -moz-border-radius: 50%;
    -ms-border-radius: 50%;
    -o-border-radius: 50%;
    border-radius: 50%;
  }

    .guide h3 a.toggle {
      float: right;
      font-size: 10px;
      color: #ccc;
      font-weight: normal;
      text-decoration: none;
      text-transform: uppercase;
    }

  .guide .description{
    max-width: 650px;
  }

    .guide .description ol,
    .guide .description ul {
      margin-top: 0;
      margin-bottom: 10px;
    }

    .guide .description p {
      margin: 0 0 10px 0;
    }

  .guide .hidden { display: none; }

  .guide .preview iframe {
    width: 100%;
    border: none;
    margin: 0;
    background: #fff;
    -webkit-border-radius: 3px;
    -moz-border-radius: 3px;
    -ms-border-radius: 3px;
    -o-border-radius: 3px;
    border-radius: 3px;
    -webkit-box-shadow: 0px 2px 2px 0px rgba(0, 0, 0, 0.15);
    -moz-box-shadow: 0px 2px 2px 0px rgba(0, 0, 0, 0.15);
    box-shadow: 0px 2px 2px 0px rgba(0, 0, 0, 0.15);
    border: solid 1px #c9c9c9;
  }

  .guide .source {
    min-height: 60px;
    font-size: 12px;
    line-height: 17px;
    font-family: Monaco, Menlo, Consolas, "Courier New", monospace;
    color: green;
    background-color: #f7f7f9;
    border: 1px solid #e1e1e8;
    border-radius: 5px;
    padding: 20px;
    white-space: pre;
    overflow: auto;
  }



/* misc */
.sourceJs,
.sourceCss {
  display: none;
}
</style></head><body><div class="container"><h1 class="title-large">Styleguide</h1><time class="generated">Generated at Tue Jun 10 2014 16:20:23 GMT+0200 (CEST)</time><hr class="small"><nav class="index"><ul><li>&rsaquo;<a href="#Buttons">Buttons</a><ul><li>&rsaquo;<a href="#Buttons0">Square buttons</a></li><li>&rsaquo;<a href="#Buttons1">Round buttons</a></li><li>&rsaquo;<a href="#Buttons2">Links</a></li></ul></li><li>&rsaquo;<a href="#Typography">Typography</a><ul><li>&rsaquo;<a href="#Typography0">Headings</a></li><li>&rsaquo;<a href="#Typography1">Paragraphs</a></li></ul></li></ul></nav><hr class="small"><nav class="nav"><ul><li><a href="#Buttons">&rsaquo; Buttons</a></li><li><a href="#Typography">&rsaquo; Typography</a></li></ul></nav><div id="Buttons" class="section"><h2 class="title-medium">Buttons</h2><div id="Buttons0" class="guide"><h3 class="title-small">Square buttons<a href="#" class="toggle">Toggle source</a></h3><div class="description"><p>Very pretty square buttons</p>
</div><div class="preview"><iframe scrolling="no"></iframe></div><div class="source hidden">&lt;a href=&quot;&quot; class=&quot;btn btn-small&quot;&gt;button&lt;/a&gt; &lt;a href=&quot;&quot; class=&quot;btn btn-medium&quot;&gt;button&lt;/a&gt; &lt;a href=&quot;&quot; class=&quot;btn btn-large&quot;&gt;button&lt;/a&gt;</div></div><div id="Buttons1" class="guide"><h3 class="title-small">Round buttons<a href="#" class="toggle">Toggle source</a></h3><div class="description"><p>Very pretty rounded buttons</p>
</div><div class="preview"><iframe scrolling="no"></iframe></div><div class="source hidden">&lt;a href=&quot;&quot; class=&quot;btn btn-small btn-round&quot;&gt;button&lt;/a&gt; &lt;a href=&quot;&quot; class=&quot;btn btn-medium btn-round&quot;&gt;button&lt;/a&gt; &lt;a href=&quot;&quot; class=&quot;btn btn-large btn-round&quot;&gt;button&lt;/a&gt;</div></div><div id="Buttons2" class="guide"><h3 class="title-small">Links<a href="#" class="toggle">Toggle source</a></h3><div class="description"><p>Very pretty rounded buttons</p>
</div><div class="preview"><iframe scrolling="no"></iframe></div><div class="source hidden">&lt;a href=&quot;&quot; class=&quot;btn-link&quot;&gt;button&lt;/a&gt;</div></div><hr class="small"></div><div id="Typography" class="section"><h2 class="title-medium">Typography</h2><div id="Typography0" class="guide"><h3 class="title-small">Headings<a href="#" class="toggle">Toggle source</a></h3><div class="description"><p>Very pretty headings</p>
</div><div class="preview"><iframe scrolling="no"></iframe></div><div class="source hidden">&lt;h1 class=&quot;title-large&quot;&gt;Heading&lt;/h1&gt; &lt;h1 class=&quot;title-medium&quot;&gt;Heading&lt;/h1&gt; &lt;h1 class=&quot;title-small&quot;&gt;Heading&lt;/h1&gt;</div></div><div id="Typography1" class="guide"><h3 class="title-small">Paragraphs<a href="#" class="toggle">Toggle source</a></h3><div class="description"><p>Very pretty paragraphs</p>
</div><div class="preview"><iframe scrolling="no"></iframe></div><div class="source hidden">&lt;p class=&quot;p-large&quot;&gt;Mutley, you snickering, floppy eared hound. When courage is needed, you're never around. Those medals you wear on your moth-eaten chest should be there for bungling at which you are best. So, stop that pigeon, stop that pigeon, stop that pigeon, stop that pigeon, stop that pigeon, stop that pigeon, stop that pigeon. Howwww! Nab him, jab him, tab him, grab him, stop that pigeon now.&lt;/p&gt; &lt;p class=&quot;p-medium&quot;&gt;Ten years ago a crack commando unit was sent to prison by a military court for a crime they didn't commit. These men promptly escaped from a maximum security stockade to the Los Angeles underground. Today, still wanted by the government, they survive as soldiers of fortune. If you have a problem and no one else can help, and if you can find them, maybe you can hire the A-team.&lt;/p&gt; &lt;p class=&quot;p-small&quot;&gt;There's a voice that keeps on calling me. Down the road, that's where I'll always be. Every stop I make, I make a new friend. Can't stay for long, just turn around and I'm gone again. Maybe tomorrow, I'll want to settle down, Until tomorrow, I'll just keep moving on.&lt;/p&gt;</div></div><hr class="small"></div><textarea class="sourceCss">body {
  font: 16px Verdana;
}

/***
  title: Square buttons
  section: Buttons
  description: Very pretty square buttons
  example:
    &lt;a href=&quot;&quot; class=&quot;btn btn-small&quot;&gt;button&lt;/a&gt;
    &lt;a href=&quot;&quot; class=&quot;btn btn-medium&quot;&gt;button&lt;/a&gt;
    &lt;a href=&quot;&quot; class=&quot;btn btn-large&quot;&gt;button&lt;/a&gt;
***/

.btn{
  display: inline-block;
  padding: .3em .6em;
  color: white;
  text-decoration: none;
  text-transform: uppercase;
  background-color: darkslateblue;
}
.btn:hover{
  background-color: #38306E;
}
.btn-small{
  font-size: .8em;
}
.btn-medium{
  font-size: 1em;
}
.btn-large{
  font-size: 1.3em;
}


/***
  title: Round buttons
  section: Buttons
  description: Very pretty rounded buttons
  example:
    &lt;a href=&quot;&quot; class=&quot;btn btn-small btn-round&quot;&gt;button&lt;/a&gt;
    &lt;a href=&quot;&quot; class=&quot;btn btn-medium btn-round&quot;&gt;button&lt;/a&gt;
    &lt;a href=&quot;&quot; class=&quot;btn btn-large btn-round&quot;&gt;button&lt;/a&gt;
***/

.btn-round{
  border-radius: 20px;
}


/***
  title: Links
  section: Buttons
  description: Very pretty rounded buttons
  example:
    &lt;a href=&quot;&quot; class=&quot;btn-link&quot;&gt;button&lt;/a&gt;
***/

.btn-link{
  background: none;
  color: darkslateblue;
}
.btn-link:hover{
  text-decoration: none;
}



/***
  title: Headings
  section: Typography
  description: Very pretty headings
  example:
    &lt;h1 class=&quot;title-large&quot;&gt;Heading&lt;/h1&gt;
    &lt;h1 class=&quot;title-medium&quot;&gt;Heading&lt;/h1&gt;
    &lt;h1 class=&quot;title-small&quot;&gt;Heading&lt;/h1&gt;
***/

h1, h2, h3, h4, h5{
  margin: 0 0 .5em 0;
  font-size: 1em;
  line-height: 1;
  font-weight: bold;
}
.title-large{
  font-size: 1.8em;
}
.title-medium{
  font-size: 1.4em;
}
.title-small{
  font-size: 1.1em;
}


/***
  title: Paragraphs
  section: Typography
  description: Very pretty paragraphs
  example:
    &lt;p class=&quot;p-large&quot;&gt;Mutley, you snickering, floppy eared hound. When courage is needed, you're never around. Those medals you wear on your moth-eaten chest should be there for bungling at which you are best. So, stop that pigeon, stop that pigeon, stop that pigeon, stop that pigeon, stop that pigeon, stop that pigeon, stop that pigeon. Howwww! Nab him, jab him, tab him, grab him, stop that pigeon now.&lt;/p&gt;
    &lt;p class=&quot;p-medium&quot;&gt;Ten years ago a crack commando unit was sent to prison by a military court for a crime they didn't commit. These men promptly escaped from a maximum security stockade to the Los Angeles underground. Today, still wanted by the government, they survive as soldiers of fortune. If you have a problem and no one else can help, and if you can find them, maybe you can hire the A-team.&lt;/p&gt;
    &lt;p class=&quot;p-small&quot;&gt;There's a voice that keeps on calling me. Down the road, that's where I'll always be. Every stop I make, I make a new friend. Can't stay for long, just turn around and I'm gone again. Maybe tomorrow, I'll want to settle down, Until tomorrow, I'll just keep moving on.&lt;/p&gt;
***/


.p-large{
  font-size: 1.1em;
  line-height: 1.3;
}

.p-medium{
  font-size: .9em;
  line-height: 1.5;
}

.p-small{
  font-size: .6em;
  line-height: 1.5;
}
</textarea><textarea class="sourceJs">console.log('im inside script.js')</textarea></div><script>sourceCss = document.querySelector('textarea.sourceCss').textContent;
sourceJs = document.querySelector('textarea.sourceJs').textContent;

// forEach for nodelists...
NodeList.prototype.forEach = Array.prototype.forEach;

function GuideElement(element) {
    var preview = element.querySelector('.preview');
    var source = element.querySelector('.source');
    var iframe = element.querySelector('.preview iframe');

    if(!source) {
        return false;
    }

    var d = iframe.contentWindow.document;
    d.open();
    d.write("<!DOCTYPE html><html><head></head><body style='padding: 20px'>"+ source.textContent +"</body></html>");

    // append stylesheet
    var stylesheet = d.createElement('style');
    stylesheet.innerHTML = sourceCss;
    d.querySelector('head').appendChild(stylesheet);

    // append js
    var script = d.createElement('script');
    script.innerHTML = sourceJs;
    d.querySelector('head').appendChild(script);

    d.close();


    // toggle link
    element.querySelector("a.toggle").addEventListener("click", function(ev) {
        preview.classList.toggle('hidden');
        source.classList.toggle('hidden');
        ev.preventDefault();
    });
}

document.querySelectorAll('.guide').forEach(function(guide) {
    GuideElement(guide);
});

var iframes = document.querySelectorAll('.preview iframe');
function scaleIframes() {
    iframes.forEach(function(frame) {
        frame.height = frame.contentWindow.document.querySelector('html').scrollHeight;
    });
}

scaleIframes();
setInterval(scaleIframes, 500);
</script></body></html>